### ISSUE TITLE
Import from Local Folder EML files Errors

### ISSUE THREAD

**Issue by toomanylogins** [thundernest/import-export-tools-ng#370]
I am importing large archives of old mailboxes in EML format to local folders. The import process works without problem but the small indicator which runs during import indicates about 10% of the emails are errors for example 2300/23000.

Is there any information about what the errors are and how to fix ?
Thanks for the extension.

**Comment by cleidigh** [comment]
@toomanylogins 
Happy Holidays!
End of year EML cleanup ;-}
Clarifications : 
- id os,tb and ptng versions 
- I am unclear on the import status as you say the import process works fine, as in everything imported correctly?
- - Then are we investigating problems with error reporting?
- Have you seen any errors in the console.
- Any history importing before?

@cleidigh

**Comment by toomanylogins** [comment]
Thanks for reply. Happy New year !

Windows 10 TB 102.3.1 Your Version 12.0.1 : Maintenance Release - July 8, 2022.

I did a bit of testing to eliminate if it was sub folder and its not. Eg I have a local folder (no subfolder) with 1709 eml files however on import the TB reports 756 in folder after import.  During import the progress in footer shows EML files Imported xxx/1709. After a few hundred you get a few errors ie 450 (Errs:20)/1709.

Always imports the same 756/1709 messages.

How do I see "errors in the console" ?
Thanks

**Comment by cleidigh** [comment]
@toomanylogins 
Ok so the import status matches the result,you don't get all 1709 messages imported?
To see the console log go to Tools\Developer Tools\Error Console
Shortcut  Shift-Control-J
Clear log with trashcan, do import then check \post results.
Cheers
@cleidigh

**Comment by toomanylogins** [comment]
Morning Chris,

The errors are all the same. From the console see below. They are sequential and appear to be older messages. I was using exchange most likely version 2007.

Msg #: 1709 Err #: 953
 Folder: Inbox
 Filename: E:\email_archive\email_archive\Outlook Mailbox - sales\Inbox\message-1-88053.eml
 FirstLine Microsoft Mail Internet Headers Version 2.0

Here are the first few lines of the email with error

Microsoft Mail Internet Headers Version 2.0
X-Antispam: NO; Spamcatcher 6.0.0. Score 1
Thread-Topic: Error Message problems with pdf viewing of epc and pea
Received: from mail-fx0-f212.google.com ([209.85.220.212]) by emailserver.xxx.local with Microsoft SMTPSVC(6.0.3790.3959); Fri, 22 Jan 2010 23:27:45 +0000
Received: by fxm4 with SMTP id 4so1698293fxm.12        for <sales@xxx.co.uk>; Fri, 22 Jan 2010 15:27:36 -0800 (PST)

The error occurs at [mboximport.js:1444:11](chrome://mboximport/content/mboximport/mboximport.js)

So looks like something to do with sender address ?

More than happy to send you a few samples.

HTH not urgent.

Cheers.

**Comment by cleidigh** [comment]
@toomanylogins 
Thanks.  It might be helpful to capture the entire log and post as a text file. This will make it more readable.
@cleidigh

**Comment by cleidigh** [comment]
@toomanylogins 
Happy New Year!
Is this still live?
Do you want to continue to debug?
@cleidigh

**Comment by rakiscats** [comment]
@cleidigh Chris, I just imported 3900+ eml messages from Windows Mail (Vista Home Basic).  110 errors, of which 108 where in the Inbox (756 emails total) and they were not imported.   Thanks to your shortcut, console log says they are all error 11 - first line.  How do I find the offending emails in the original Inbox?  I am a (very very old) programmer and know HTML but I need to know what to look for in the original emails.  Thanks, Cathy

**Comment by cleidigh** [comment]
@rakiscats 
There is no such thing as very old programmers, we are just "seasoned" %-}

That is not good on the import losses so I am very interested to debug, especially as I am working furiously on v14.0.1 to get as many known issues fixed as possible.

Can you capture and paste some of these errors, I want to see exactly how they are reported. Do you have a small directory of emls that you know produce errors that we might start to look at?
@cleidigh (Lisp, C and my love for assembly language)

**Comment by rakiscats** [comment]
@cleidigh
Mystery is partially solved! There are really only 646 messages in the Inbox, but Windows Mail exports 756 messages of which 108 are 0KB. So we know why TB skips them ("empty" error) but not why they were generated. My guess at the moment is...."placeholders" for my Contacts in Windows Mail, which I did not select to export...they display as 112 entries. Whatever they are, the dates they were originally created (as stored in Windows Mail) get preserved.

But I have a new issue for you with Import.  I discovered that a lot of my imported Eudora Pro messages (I imported these into Windows Mail years ago) do not display the Date properly in TB.  The DATE/TIME of my original import is all I see in the Date tab in the new TB folders.  No Date/Time displays when I open the messages.  Here is a sample source header.

From - Wed Nov 01 11:33:16 2023
X-Mozilla-Status: 0001
X-Mozilla-Status2: 00000000
Received: ; Thu Oct 28 17:00:38 2004 -0400
X-Priority: 3
X-MSMail-Priority: Normal
To: <me>
From: "Cathy Galfo" <rakiscats@earthlink.net>
X-MimeOLE: Produced By Microsoft MimeOLE V6.0.6001.18049
Subject: Yamir being shown by Sue again.- GC (& maybe DbGC?)
Cc: 
Bcc: 
X-Attachments: 
In-Reply-To: 
References: 
Message-ID: <4.2.0.58.20041028165532.0094d5f0@mail.earthlink.net>
X-Eudora-Signature: <<No Default>>

Here is a sample header from an old email that imported properly:

From - Wed Nov 01 11:33:16 2023
X-Mozilla-Status: 0001
X-Mozilla-Status2: 00000000
Status:  U
Content-Transfer-Encoding: 7bit
Return-Path: <TOKIRA@aol.com>
X-Priority: 3
X-MSMail-Priority: Normal
Received: from imo-d21.mx.aol.com ([205.188.144.207]) by skylark (EarthLink SMTP Server) with SMTP id 1b6QSx4G33NZFjw0 for <rakiscats@earthlink.net>; Fri, 26 Mar 2004 04:56:05 -0800 (PST)
X-MimeOLE: Produced By Microsoft MimeOLE V6.0.6001.18049
Received: from TOKIRA@aol.com by imo-d21.mx.aol.com (mail_out_v37_r1.2.) id f.1ed.1c5e6a05 (4560) for <rakiscats@earthlink.net>; Fri, 26 Mar 2004 07:55:55 -0500 (EST)
From: <TOKIRA@aol.com>
Message-ID: <1ed.1c5e6a05.2d958259@aol.com>
Date: Fri, 26 Mar 2004 07:55:53 EST
Subject: funny Charly
To: <rakiscats@earthlink.net>
MIME-Version: 1.0
Content-Type: text/plain;
	charset="iso-8859-1"
X-Mailer: AOL 4.0 for Mac - Post-GM sub 147

Any chance of fixing this problem so I can re-import the old emails with the correct date?
Thanks again.  @rakiscats

**Comment by cleidigh** [comment]
@rakiscats 
Running slow today, but wanted to fire off a couple questions while I check on some things.

First it's really important to describe the origin of whatever you post as well as if you have done any edits such as far privacy, just so I know.

The top example  (ex1) looks like you obtained it by viewing its source with Control-U or View Source POST IMPORT
Is this correct? My assumption is based on the first line which should not be part of a correct eml message. And its the date you must have impoted added by TB.

This message has no Date: header, but does have a Received: header which can be used as the message date, however, this has a semicolon which is totally wacky.

The (ex2) looks like maybe you got from the pre imported message given the first line, but I can't be sure. Ex2 has a proper Date: header and all else looks normal.

So my first ask is if I can get ex1 & ex2 PRE import with all headers.
If you prefer you can send to my test account : 
test1@kokkini.net
I won't reply from there, but here.

Sorry for the long ramblings, but there are so many things that can be of issue and best you get where I am going. Most likely it's the non standard date handling. If you can also check another bad message to see if it has a semicolon in Received: 

One way or another we will get it.
@cleidigh

**Comment by rakiscats** [comment]
@cleidigh
I agree that the first "Received" message header has a strange looking date....
Both headers are from POST import source code in TB.
I can try to get PRE import of the same messages emailed to you from this Yahoo account.
I think about half of my EudoraPro emails have the date problem in TB, but Windows Mail was able to decipher them (wonders never cease <g>).
@rakiscats

**Comment by cleidigh** [comment]
@rakiscats 
That would be great.
In the meantime I looked at the very old code (pre dating myself) and I am not sure yet where things go wrong, but with the pre import data should be able to reproduce.
Can you also post the console output ?
@cleidigh
